File: 8_BaseExpansionToDecimal/8.rb
def toDecimal(binary)
    num = 0
    j = 0
    len = binary.length
    (len - 1).downto(0) do |i|
        if binary[i] == '1'
            num += 2 ** j
        end
        j += 1
    end
    return num
end


print "Escribe el número en binario: "
binario = gets.chomp.to_s
puts "El número es: #{toDecimal(binario)}"

File: 8_BaseExpansionToDecimal/8.py
def convertirADecimal(binario):
    num = 0
    j = 0
    for i in range(len(binario)):
        if(binario[-(i+1)] == '1'):
            num += 2 ** j
        j = j + 1
    return num

binario = str(input("Escribe el número: "))
print("El número es %d" % convertirADecimal(binario))
